### PR TITLE
avc278 - 1.04 - Javascript

### PR DESCRIPTION
**Pull request by avc278**
Added my solution to the Chapter 1 Problem 4 folder.

### CODE REVIEW

File: JavaScript/chapter01/p04_palinperm/avc278.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ *
+ * Runtime: O(N)
+ * Space:   O(N)
+ *
+ */
+const palindromePerm = (str) => {
+  const letterFreq = {};
+  let length = 0;
+  for (const letter of str) {
+    if (letter !== " ") {
```
Comment (lhchavez) [inline JavaScript/chapter01/p04_palinperm/avc278.js:26]: small optional readability suggestion: here (and in L37, L38), the [guard clause pattern](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) can be used to reduce nesting and improve readability:

```suggestion
    if (letter === " ") continue;  // or in the next line.
```

File: JavaScript/chapter01/p04_palinperm/avc278.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        (letterFreq[letter.toLowerCase()] || 0) + 1;
+      length += 1;
+    }
+  }
+
+  const keys = Object.keys(letterFreq);
+
+  let slack = length % 2 == 1;
+  for (const key of keys) {
+    if (letterFreq[key] % 2 !== 0) {
```
Comment (lhchavez) [inline JavaScript/chapter01/p04_palinperm/avc278.js:37]: given that `key` is only used to get the value of the key/value pair, it's probably better to express that more directly with [`Object.values()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values):

```suggestion
  for (const letterFreq of Object.values(letterFreqs)) {  // assuming L23 is renamed to be plural.
    if (letterFreq % 2 === 0) continue;
```

File: JavaScript/chapter01/p04_palinperm/avc278.js (hunk 1 above)
Comment (avc278) [reply JavaScript/chapter01/p04_palinperm/avc278.js:26]: oops... I actually knew this one :P

Comment (avc278) [reply JavaScript/chapter01/p04_palinperm/avc278.js]: Completely forgot to update this one. Should be fixed now

**Pull request merged**